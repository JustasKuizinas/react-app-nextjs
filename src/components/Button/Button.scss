.btn {
  text-decoration: none;
  display: inline-block;
  &.-primary {
    height: 60px;
    text-transform: uppercase;
    color: #fff;
    background-color: var(--color-primary);
    border-radius: 5px;
    font-size: 2rem;
    padding: 0 20px;
  }
 
  &.-secondary {
    background-color: rgba(83, 83, 83, 0.8);
    text-transform: uppercase;
    color: var(--color-primary);
    border-radius: 5px;
    height: 40px;
    padding: 0 20px; 
    font-size: 1.8rem;
  }
  &.-trenary {
    border-radius: 3px;
    height: 50px;
    padding: 0 20px;
    font-size: 1.8rem;
    background-color: var(--color-primary);
    text-transform: uppercase;
    color: #fff;
  }
  &.-outline {
    border: solid 2px var(--color-primary);
    color: var(--color-primary);
    border-radius: 3px;
    height: 50px;
    line-height: 50px;
    padding: 0 20px;
    font-size: 1.8rem;
    background-color: transparent;
    text-transform: uppercase;
  }
}
