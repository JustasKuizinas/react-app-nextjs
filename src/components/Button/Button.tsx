import * as React from 'react';
import PropTypes from 'prop-types';
// import './Button.scss';

const Button: React.FC<any> = props => {
  return (
    <>
      <button
        type={props.type}
        className={'btn ' + props.style}
        onClick={props.onClick}
      >
        {props.children}
      </button>
    
    </> 
  );
};

Button.propTypes = {};

Button.defaultProps = {
  type: 'button',
  style: '',
  onClick: () => {},
};

export default Button;
