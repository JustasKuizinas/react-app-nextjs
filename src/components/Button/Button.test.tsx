import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import Button from './Button';

describe('Button', () => {
  test('Contains text', () => {
    // const app = shallow(<Button>Button</Button>);
    // expect(app.find('#app').text()).toEqual('Hello');
    const props = {
      style: '-primary',
      onSeonClick: (value: boolean) => {
        return;
      },
    };
    const tree = renderer.create(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
