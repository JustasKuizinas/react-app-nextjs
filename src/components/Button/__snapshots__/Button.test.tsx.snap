// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Contains text 1`] = `
<button
  className="btn -primary"
  onClick={[Function]}
  type="button"
/>
`;
